File: chain-of-responsibility/python/custom_handler.py
from abstract_handler import AbstractHandler
from image import Image


class CustomHandler(AbstractHandler):
    """
    Herhangi bir tipteki dosyayı dönüştüren sınıf.
    UML diyagramındaki ConcreteHandler sınıfına denk gelmektedir.
    """

    def handle_request(self, image: Image):
        if image.extension == "OtherExtension":
            # Herhangi bir tipteki dosyanın işlemine ait dönüştürme kodları işlemleri.
            print("OtherExtension to PNG")
        else:
            return super().handle_request(image)

File: chain-of-responsibility/php/CustomHandler.php
<?php

// Herhangi bir tipteki dosyayı dönüştüren sınıf.
// UML diyagramındaki ConcreteHandler sınıfına denk gelmektedir.
class CustomHandler extends BaseHandler
{
    public function handleRequest(Image $image): bool
    {
        if ($image->extension == "OtherExtension") {
            echo "OtherExtension to PNG";
            // Herhangi bir tipteki dosyanın işlemine ait dönüştürme kodları işlemleri.

            return true;
        }

        return false;
    }
}
